test/functional/aws_elasticache_resources_exist.go

```diff
@@ -0,0 +1,104 @@
+		redis := &crov1.Redis{}
+		if err := ctx.Client.Get(goContext, types.NamespacedName{Namespace: common.RHMIOperatorNamespace, Name: r}, redis); err != nil {
+			testErrors = append(testErrors, fmt.Sprintf("\nfailed to find %s redis cr : %v", r, err))
+		}
+		// ensure phase is completed
+		if redis.Status.Phase != croTypes.PhaseComplete {
+			testErrors = append(testErrors, fmt.Sprintf("\nfound %s redis not ready with phase: %s, message: %s", r, redis.Status.Phase, redis.Status.Message))
+		}
+		// return resource id
+		resourceID, err := GetCROAnnotation(redis)
+		if err != nil {
+			testErrors = append(testErrors, fmt.Sprintf("\n%s redis does not contain a resource id annotation: %v", r, err))
+		}
+		// populate the array
+		elasticacheResourceIDs = append(elasticacheResourceIDs, resourceID)
+	}
+
+	if len(testErrors) != 0 {
+		t.Fatalf("test cro redis exists failed with the following errors : %s", testErrors)
+	}
+
+	// create AWS session
+	sess, err := resources.CreateAWSSession(goContext, ctx.Client)
+	if err != nil {
+		t.Fatalf("failed to create aws session: %v", err)
+	}
+
+	// create new elasticache api with retrieved session
+	elasticacheapi := elasticache.New(sess)
+
+	// iterate through returned resource ID's
+	for _, resourceID := range elasticacheResourceIDs {
+		//get elasticache resources through new elasticacheapi
+		foundElasticacheReplicationGroups, err := elasticacheapi.DescribeReplicationGroups(&elasticache.DescribeReplicationGroupsInput{
+			ReplicationGroupId: aws.String(resourceID),
+		})
+		if err != nil {
+			testErrors = append(testErrors, fmt.Sprintf("failed to get %s elasticache replicationgroups with error : %v", resourceID, err))
+			continue
+		}
+
+		replicationGroup := foundElasticacheReplicationGroups.ReplicationGroups[0]
+		nodeGroup := replicationGroup.NodeGroups[0]
+
+		// perform checks to verify state is as expected
+		if err := verifyMultiAZ(nodeGroup.NodeGroupMembers); err != nil {
+			testErrors = append(testErrors, fmt.Sprintf("elasticache resource %s multiAZ failure %v", resourceID, err))
+		}
+		if !aws.BoolValue(replicationGroup.AtRestEncryptionEnabled) {
+			testErrors = append(testErrors, fmt.Sprintf("elasticache resource %s does not have encryption enabled", resourceID))
+		}
+		if replicationGroup.SnapshotWindow == nil {
+			testErrors = append(testErrors, fmt.Sprintf("elasticache resource %s does not have automatic snapshotting enabled", resourceID))
+		}
+	}
+
+	if len(testErrors) != 0 {
+		t.Fatalf("test elasticache exists failed with the following errors : %s", testErrors)
+	}
+
+}
+
+// helper method for verifying nodes are in different availability zones
+func verifyMultiAZ(member []*elasticache.NodeGroupMember) error {
+	if len(member) < 1 {
+		return errors.New("insufficient number of nodes in elasticache group")
+	}
+	if member[0].PreferredAvailabilityZone == member[1].PreferredAvailabilityZone {
+		return errors.New("replication group is not multiAZ")
+	}
+	return nil
+}
```

Comment:
Not a huge improvement, but may be slightly cleaner to change the function signature to return a `bool` instead: 

```suggestion
func verifyMultiAZ(members []*elasticache.NodeGroupMember) bool {
	return members[0].PreferredAvailabilityZone == members[1].PreferredAvailabilityZone
}
```
Then we wouldn't need the error check when calling the function either.

Reply:
I Think leaving it to return an error is better, due to the fact that when It does fail, at least we've a specific description of why it failed as opposed to just getting a "Fail" and being left in the dark as to why

Reply:
You could always return the same error above when you call the function